### PR TITLE
new cbor encoder

### PR DESCRIPTION
**Pull request by nyaxt**

### CODE REVIEW

File: go/signedexchange/cbor/encoder.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+		nfollow = 4
+	default:
+		ai = 27
+		nfollow = 8
+	}
+
+	encoded := make([]byte, 1+nfollow)
+	encoded[0] = byte(t) | ai
+	for i := nfollow - 1; i >= 0; i-- {
+		encoded[i+1] = byte(n & 0xff)
```
Comment (hajimehoshi) [inline go/signedexchange/cbor/encoder.go]: byte(n) is enough?

File: go/signedexchange/cbor/encoder.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+import (
+	"io"
+)
+
+type Encoder struct {
+	w io.Writer
+}
+
+func (e Encoder) encodeTypedUInt(t Type, n uint64) error {
```
Comment (hajimehoshi) [inline go/signedexchange/cbor/encoder.go]: (e *Encoder)

File: go/signedexchange/cbor/encoder.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		n >>= 8
+	}
+
+	if _, err := e.w.Write(encoded); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (e Encoder) EncodeUInt(n uint64) error {
```
Comment (hajimehoshi) [inline go/signedexchange/cbor/encoder.go]: ditto

File: go/signedexchange/cbor/encoder.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	return e.encodeTypedUInt(TypePosInt, n)
+}
+
+func (e Encoder) EncodeInt(n int64) error {
```
Comment (hajimehoshi) [inline go/signedexchange/cbor/encoder.go]: ditto

File: go/signedexchange/cbor/types.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+//    since they don't fit in a 2's-complement int64.
+//  * Floating-point numbers
+//  * Indefinite-length encodings.
+//  * Parsing
+package cbor
+
+type Type byte
+
+const (
+	TypePosInt Type = iota << 5
```
Comment (hajimehoshi) [inline go/signedexchange/cbor/types.go]: This enum values would be 0, 32, 64, 96, ... that exceeds `byte` values. Is this expected?

Comment (nyaxt) [reply go/signedexchange/cbor/types.go]: Yeah, probably its better to be explicit here.

**Comment by nyaxt** [comment]
@hajimehoshi Thanks. Addressed your comments.

File: go/signedexchange/cbor/types.go (hunk 5 above)
Comment (hajimehoshi) [reply go/signedexchange/cbor/types.go]: Ah, as maximum values are 0xc0, this is not a problem. Please ignore the above comment.

**Comment by hajimehoshi** [comment]
lgtm

**Pull request merged**